### PR TITLE
Added JobRun_Silver in buildClusterStateFact for Cluster E…

### LINKED ISSUE #1020: Impute automated clusters termination as a last resort
There have been a few issues with the platform where we might miss the terminating event for an automated cluster. Can we investigate if it would be possible to impute the cluster termination from the job termination timestamp?

### PR DESCRIPTION
**Pull request by souravbaner-da**
Added Change for imputation for clusters termination.
Close #1020

### CODE REVIEW

**Review by aman-db — APPROVED** [review]

**Comment by dougspadotto** [comment]
Hi. Is there an estimate on when this will be merged?

**Comment by gueniai** [comment]
@dougspadotto We're targetting first week of December

File: src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala [hunk 1]
```diff
@@ -402,8 +403,21 @@ trait GoldTransforms extends SparkSessionWrapper {
       (driverComputeCost + driverDBUCost + workerComputeCost + workerDBUCost).alias("total_cost")
     )
 
-    clusterPotential
+    val clsfDF = clusterPotential
       .select(clusterStateFactCols: _*)
+
+    val clsfDF1 =clsfDF.filter('state === "TERMINATING").filter('isAutomated === true).filter('days_in_state > 2)
```
Comment (gueniai) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala:409]: This handles one of the corner cases, but doesn't help us in cases where the job finishes and we're missing the TERMINATING event altogether. 
In my opinion, we should approach it from the the other side, find the job runs that used an automated cluster and terminated, then find those on the list for clsf and mark them as terminated

**Review by gueniai — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala [hunk 2]
```diff
@@ -1020,13 +1023,44 @@ trait SilverTransforms extends SparkSessionWrapper {
 
     val invalidEventChain = lead('runningSwitch, 1).over(stateUnboundW).isNotNull && lead('runningSwitch, 1)
       .over(stateUnboundW) === lead('previousSwitch, 1).over(stateUnboundW)
-    val clusterEventsBaseline = clusterEventsDF
+
+
+    val clusterEventsDF1 = clusterEventsDF
```
Comment (gueniai) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1028]: I know these inner variables are hard to name but let's find something more descriptive than xyzDF1. Longer variable names are fine

**Review by gueniai — CHANGES_REQUESTED** [review]
Minor nitpick around more declarative variable names, I think this is almost done. Did you do a second run to ensure the terminated_impute was respected as a terminal event and the cluster's events weren't progressed anymore? (I couldn't find it in the proof)

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala (hunk 2 above)
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala]: Done

**Review by gueniai — APPROVED** [review]

**Pull request merged**